### ISSUE TITLE
dataForColumn: returns NULL when the column contains zero-length NSData

### ISSUE THREAD

**Issue by jmatx** [ccgus/fmdb#350]
If I store  `[NSData data]`  (a zero-length NSData value) into a column, then
`dataForColumn:` returns NULL.  It should return a zero-length NSData.

**Comment by robertmryan** [comment]
Good point. It really should distinguish between `NULL` values and zero length `BLOB` objects.

By way of background, the [`sqlite3_column_blob` documentation](https://www.sqlite.org/c3ref/column_blob.html) tells us:

> The return value from `sqlite3_column_blob()` for a zero-length BLOB is a `NULL` pointer.

And the `dataForColumnIndex` method (which is called by the `dataForColumn` method) detects this situation. This method, however, returns `nil` in this case and there's a compelling argument for it returning `[NSData data]` instead.

The tactical issue is that FMDB has a large install-base, and we probably don't want to just change this behavior for fear of breaking existing code. There may be people out there who have written code looking for `nil`, so we should be circumspect about changing this behavior.

If this behavior is changed, I'd suggest deferring this until a major release of the library which probably will have a number of non-backward compatible changes. This should not be a change that is introduced in a minor update of the library.

**Comment by ccgus** [comment]
I'll consider this for 3.0
